### PR TITLE
Numeric Support

### PR DESCRIPTION
**Pull request by Shi-Raida**
- Can have integers with only one bound from the UPF
- Can use action parameters as action cost

### CODE REVIEW

File: planning/grpc/server/src/chronicles.rs [hunk 1]
```diff
@@ -895,13 +895,14 @@ impl<'a> ChronicleFactory<'a> {
...
-        ensure!(kind(cost)? == ExpressionKind::Constant);
-        ensure!(cost.r#type == "up:integer");
-        let cost = match cost.atom.as_ref().unwrap().content.as_ref().unwrap() {
-            Content::Int(i) => *i as IntCst,
-            _ => bail!("Unexpected cost type."),
+        let value = self.reify(cost, None)?;
+        match value {
+            Atom::Int(i) => {
+                self.chronicle.cost = Some(i);
+                self.chronicle.constraints.push(Constraint::leq(IAtom::ZERO, i));
```
Comment (arbimo) [inline planning/grpc/server/src/chronicles.rs:902]: This seems incorrect.
Previously, we would disallow user to use negative cost and warn them when encoding the problem.
Here you allow them to do this but would make the problem unsolvable (or at least remove some of the expected solution).

We should check whether there is still a reason to disallow negative costs.

File: planning/planners/src/encode.rs [hunk 2]
```diff
@@ -438,8 +437,10 @@ pub fn add_metric(pb: &FiniteProblem, model: &mut Model, metric: Metric) -> IAto
             let action_costs: Vec<LinearTerm> = costs
                 .iter()
                 .map(|&(ch_id, p, cost)| {
+                    let bounds = model.domain_of(*cost);
+                    assert!(bounds.0 >= 0, "A chronicle could have a negative cost");
                     model
-                        .new_optional_ivar(cost, cost, p, Container::Instance(ch_id).var(VarType::Cost))
+                        .new_optional_ivar(bounds.0, bounds.1, p, Container::Instance(ch_id).var(VarType::Cost))
```
Comment (arbimo) [inline planning/planners/src/encode.rs:443]: Incorrect. The cost you use in the linear term is a new variable that is not linked to actual cost variable (just required to be within its *initial* bounds). 

Why not use the cost variable directly ?

**Review by arbimo — CHANGES_REQUESTED** [review]

File: planning/grpc/server/src/chronicles.rs (hunk 1 above)
Comment (Shi-Raida) [reply planning/grpc/server/src/chronicles.rs]: The constraint has been removed.
The negative cost are now possible for an action.
See cad2fa0abf70341ed6e834d278ae7b4c2ebaffd1.

File: planning/planners/src/encode.rs (hunk 2 above)
Comment (Shi-Raida) [reply planning/planners/src/encode.rs]: Fixed in 031365a72a51bf8c03511dff2126cad5a7c5b9fc.

**Pull request closed without merging**